game-core/src/main/java/games/strategy/engine/data/gameparser/XmlGameElementMapper.java

```diff
@@ -175,8 +175,7 @@ private static void handleMissingObject(final String objectTypeName, final Strin
       final GameData gameData) {
     checkNotNull(typeName);
 
-    final String normalizedTypeName =
-        typeName.replaceAll("^games\\.strategy\\.triplea\\.attachments\\.", "");
+    final String normalizedTypeName = typeName.replaceAll("^.*\\.", "");
```

Comment:
We'll need to test and fix that if needed.  Having an 'or' regex is not really ideal, but we could do it. IMO the java-class designation ideally would be completely removed, but we'll need to support it going forward. Perhaps we could do just straight up hardcoded mappings. We still have reflection going on to create classes by name, an explicit mapping with explicit construction code is I think where I'd like to go (and then couple that with updating the XML so that java classes are inferred on the backend and no longer needed in the XML).